### PR TITLE
Beaker enhancement

### PR DESCRIPTION
**Pull request by p3ck**
Includes the following:

reserve_duration: can be specified so that resources can be held onto for longer than the default 24 hours
#612 is supported in this PR
#611  is supported in this PR.

### CODE REVIEW

**Comment by herlo** [comment]
I think I see what was going on here. You had 'false' for both of those in the interim before finding the excludes option. [This](https://github.com/CentOS-PaaS-SIG/linchpin/pull/614/commits/013a0169ae027d8ff65659a310f7ff83201b7573) is why I thought that was the case.

**Comment by p3ck** [comment]
any idea why travis exited non-zero?  the test cases look good.

**Comment by anascko** [comment]
I apologize for the intervention
But we tested your changes regarding family and tags 
and have some issue:
Is it the correct format of schema.json for the tag?
https://github.com/beaker-project/beaker/blob/develop/Client/src/bkr/client/__init__.py#L218-L222
And if we used family instead of a distro and tag not specified then beaker will be used default tag

**Comment by p3ck** [comment]
I have tested this morning using the following in the schema
          recipesets:
            - family: "Fedora28"
              tags:
                - released

Which works as expected.  The latest schema update will require family or distro.

**Comment by anascko** [comment]
It works because it uses default tag, 
if you looked on beaker xml of this job, you don't find your provided tag

**Comment by p3ck** [comment]
@anascko I don't understand your statement.  Here is the xml snippet from the job I submitted via linchpin.
`      <distroRequires>
        <and>
          <distro_family op="=" value="Fedora28"/>
          <distro_tag op="=" value="released"/>
          <distro_variant op="=" value="Server"/>
          <distro_arch op="=" value="x86_64"/>
        </and>
      </distroRequires>`

You can see the distro_tag is included.  I agree it would have been missing before my last commit.  So please update and try again.

**Comment by anascko** [comment]
@p3ck Thanks,
I apologize, I didn't update the local repository with last commit afc0209773da56f215928637074479a762b297e7, that is why I still had a problem
Now tag works, thanks for the fix

**Comment by herlo** [comment]
@p3ck  Travis CI failed here: https://travis-ci.org/CentOS-PaaS-SIG/linchpin/builds/400885093#L645

File: config/Dockerfiles/tests.d/beaker/02_beaker_distro [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+TMP_FILE=$(mktemp)
+
+function clean_up {
+    set +e
+    linchpin -w . -p "${PINFILE}" -v destroy "${DTARGET}" "${FTARGET}"
+}
+trap clean_up EXIT SIGHUP SIGINT SIGTERM
+
+pushd docs/source/examples/workspace
+linchpin -w . -p "${PINFILE}" -v up "${DTARGET}" "${FTARGET}"
```
Comment (herlo) [inline config/Dockerfiles/tests.d/beaker/02_beaker_distro]: How is this test different from https://github.com/p3ck/linchpin/blob/23729fbe23f430fb0e9f4c4e28a5b1bc59009a77/config/Dockerfiles/tests.d/general/01_general ?

Comment (herlo) [reply config/Dockerfiles/tests.d/beaker/02_beaker_distro]: @p3ck nvm, I see the target data.

**Comment by p3ck** [comment]
[merge]

**Comment by p3ck** [comment]
#615 is fixed by this PR

**Pull request merged**